specifyweb/frontend/js_src/lib/components/QueryComboBox/index.tsx

```diff
@@ -352,6 +355,7 @@ export function QueryComboBox({
     hasTablePermission(field.relatedTable.name, 'create');
 
   const isReadOnly = React.useContext(ReadOnlyContext);
+  const [temporaryReadOnly, setTemporaryReadOnly] = useTriggerState(isReadOnly);
```

Comment:
the value of this is only used when state.type === 'ViewResourceState', so why not modify the state's ViewResourceState state to add a prop like `isReadOnly: boolean`?
that fixes the issue with temporaryReadOnly existing even when it's not needed (possible source of bugs)
additionally, if two states are updated at the same time, then they should be merged into one state to avoid bugs where two states get out of sync/one state is updated before the other possibly causing UI to work incorrectly
(@realVinayak had this issue several times on the stats page)